### PR TITLE
28 auth unit testing

### PR DESCRIPTION
**Pull request by lisovyk**
Rewrote tests for registration/model

### CODE REVIEW

File: moninag/tests/unittests/registration/test_model.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+        user = CustomUser()
+        user = user.get_by_id(2)
+        self.assertEqual(user.id, 2)
+
+    def test_get_by_id_false(self):
+        """Check if we get none if passed id was not found"""
+
+        user = CustomUser()
+        self.assertEqual(user.get_by_id(44444), None)
```
Comment (lhalam) [inline moninag/tests/unittests/registration/test_model.py]: use assertIsNone()

```
user = CustomUser.get_by_id(44444)
self.assertIsNone(user)
```

File: moninag/tests/unittests/registration/test_model.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+        user = CustomUser.objects.get(email="test@test.test")
+        user.update(first_name="UpdatedName", second_name="UpdatedSecondName")
+        self.assertEqual(user.first_name, "UpdatedName")
+        self.assertEqual(user.second_name, "UpdatedSecondName")
+
+    def test_get_by_id(self):
+        """Check if we can get the user by id"""
+
+        user = CustomUser()
```
Comment (lhalam) [inline moninag/tests/unittests/registration/test_model.py]:
```
user = CustomUser()
user = user.get_by_id(2)
```
to
```
user = CustomUser.get_by_id(2)
```

**Review by lhalam — CHANGES_REQUESTED** [review]

**Comment by hutsulyk** [comment]
Rebase with dev :slightly_smiling_face:

File: moninag/tests/unittests/registration/test_model.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        CustomUser.objects.create(first_name="TestName",
+                                  second_name="TestSecondName",
+                                  email="test@test.test",
+                                  avatar="avatar")
+
+    def test_full_name(self):
+        """Ensure that get_full_name method works properly"""
+
+        user = CustomUser.objects.get(email="test@test.test")
+        self.assertEqual(user.get_full_name(), "TestName TestSecondName")
```
Comment (hutsulyk) [inline moninag/tests/unittests/registration/test_model.py]: Test failed
```AttributeError: 'CustomUser' object has no attribute 'last_name'```

File: moninag/tests/unittests/registration/test_model.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    def test_save(self):
+        """Ensure that save() method works properly, changing email address to lower()"""
+
+        user = CustomUser()
+        user.first_name = "Firstname"
+        user.second_name = "Secondname"
+        user.email = "TESTmail@test.so"
+        user.avatar = "avatartest"
+        user.save()
```
Comment (lhalam) [inline moninag/tests/unittests/registration/test_model.py:32]:
```
user.save()
expected = Server.objects.get(id=user.id)
self.assertEqual(user, expected)
```

File: moninag/tests/unittests/registration/test_model.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        actual_user = CustomUser.objects.get(email="test@test.test")
+        self.assertEqual(actual_user.first_name, "UpdatedName")
+
+    def test_update(self):
+        """Ensure that we can update the first and second names"""
+
+        user = CustomUser.objects.get(email="test@test.test")
+        user.update(first_name="UpdatedName", second_name="UpdatedSecondName")
+        self.assertEqual(user.first_name, "UpdatedName")
```
Comment (lhalam) [inline moninag/tests/unittests/registration/test_model.py]:
```
expected_user =  CustomUser.objects.get(email="test@test.test")
self.assertEqual(expected_user .first_name, "UpdatedName")
```

File: moninag/tests/unittests/registration/test_model.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
+from django.test import TestCase
+from registration.models import CustomUser
```
Comment (ghost) [inline moninag/tests/unittests/registration/test_model.py]: proper import format (separate 3rd party imports from our project imports)

File: moninag/tests/unittests/registration/test_model.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        user = CustomUser.objects.get(email="test@test.test")
+        user.update(first_name="UpdatedName", second_name="UpdatedSecondName")
+        self.assertEqual(user.first_name, "UpdatedName")
+        self.assertEqual(user.second_name, "UpdatedSecondName")
+
+    def test_get_by_id(self):
+        """Check if we can get the user by id"""
+
+        user = CustomUser.get_by_id(2)
+        self.assertEqual(user.id, 2)
```
Comment (ghost) [inline moninag/tests/unittests/registration/test_model.py]: i would suggest to compare entire users, not only the id. (CustomUser.objects.get(id=id and CustomUser.get_by_id(2))

**Review by ghost — CHANGES_REQUESTED** [review]

File: moninag/tests/unittests/registration/test_model.py (hunk 3 above)
Comment (lisovyk) [reply moninag/tests/unittests/registration/test_model.py]: There is a wrong attribute in model. I will change that with the next commit.

**Review by lhalam — APPROVED** [review]

**Review by hutsulyk — APPROVED** [review]

**Review by ghost — APPROVED** [review]
seems to be OK

**Pull request merged**